src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -275,26 +275,11 @@ public Response getPoliciesForCustomer() {
             Pager pager = PagingUtils.extractPager(uriInfo);
             page = Policy.pagePoliciesForCustomer(entityManager, user.getAccount(), pager);
 
-            // Get lastTriggeredTime and add it to the policies
-            String idsToFetch = page.stream().map(p -> p.id.toString()).collect(Collectors.joining(","));
-            try {
-                List<Trigger> list = engine.findTriggersById(idsToFetch, user.getAccount());
-
-                // Put the lastTriggered time (= creation time ) of the alert in a triggerId-alert map
-                Map<String, Long> idTimeMap = new HashMap<>(page.size());
-                for (Trigger trigger : list) {
-                    long lastTriggered = getLastTriggeredFromTriggerLifecycle(trigger);
-                    idTimeMap.put(trigger.id, lastTriggered);
+            for (Policy policy : page) {
+                Long lastTriggerTime = policiesHistoryRepository.getLastTriggerTime(user.getAccount(), policy.id);
+                if (lastTriggerTime != null) {
+                    policy.setLastTriggered(lastTriggerTime);
```

Comment:
I will get rid of that iteration later in favor of a SQL join.

Reply:
or just `IN ( policyId1, policyId2, ...)`

Reply:
You're right. I wrote `join` without really looking at what the other query does :smile: